py_neuromodulation/nm_stream.py

```diff
@@ -117,42 +129,58 @@ def _run(
 
             generator = self.lsl_stream.get_next_batch()
 
-        l_features: list[dict] = []
-        last_time = None
+        prev_batch_end = 0
 
         while True:
             next_item = next(generator, None)
 
             if next_item is not None:
-                time_, data_batch = next_item
+                timestamps, data_batch = next_item
             else:
                 break
 
             if data_batch is None:
                 break
-            feature_dict = self.data_processor.process(
-                data_batch.astype(np.float64)
+
+            feature_dict = self.data_processor.process(data_batch)
+
+            this_batch_end = timestamps[-1]
+            batch_length = this_batch_end - prev_batch_end
+            logger.debug(
+                f"{batch_length:.3f} seconds of new data processed",
             )
-            if is_stream_lsl:
-                feature_dict["time"] = time_[-1]
-                if self.verbose:
-                    if last_time is not None:
-                        logger.debug("%.3f seconds of new data processed", time_[-1] - last_time)
-                    last_time = time_[-1]
-            else:
-                feature_dict["time"] = np.ceil(time_[-1] * 1000 +1 ).astype(int)
-                logger.info("Time: %.2f", feature_dict["time"]/1000)
-            
+
+            feature_dict["time"] = (
+                batch_length if is_stream_lsl else np.ceil(this_batch_end * 1000 + 1)
+            )
```

Comment:
I'm not entirely sure this is right for 2 reasons:
 - the "time" field should be consistent between file streams and lsl_streams, not having a duration for one and a time point for the other
 - Not sure if the units are consistent between both types of streams right now